util/src/sha3.rs

```diff
@@ -68,6 +68,24 @@ impl<T> Hashable for T where T: AsRef<[u8]> {
 	}
 }
 
+impl<T> Hashable for [T] where T: Hashable {
+	fn sha3(&self) -> H256 {
```

Comment:
This implementation seems strange: intuitively I'm not sure what it would mean to call
`vec!["a, "b", "c"].sha3()`, but I don't think I would expect it to be 

`sha3("a") ^ sha3("b") ^ sha3("c") ^ sha3("")`.

Why not just replace any invocation of this with
`things_to_hash.into_iter().map(|x| x.sha3()).fold(Default::default(), BitXor::bitxor)`?